### PR TITLE
KAFKA-348 - SSLContext Config added

### PR DESCRIPTION
**Pull request by bdesert**
Adding SSL Support on connector level.

[KAFKA-348](https://jira.mongodb.org/browse/KAFKA-348)

Current state:
In order to support SSL, four ssl parameters should be defined on a system level.
It is described [in documentation](https://www.mongodb.com/docs/kafka-connector/current/security-and-authentication/tls-and-x509/#add-credentials-to-the-connector).

In managed environments, like MSK Connect, this is not possible as a user doesn't have an access to the OS.

New Feature:
I'm introducing four new connector properties:

```
connection.ssl.truststore=<your path to truststore>
connection.ssl.truststorePassword=<your truststore password>
connection.ssl.keystore=<your path to keystore>
connection.ssl.keystorePassword=<your keystore password>
```

These values will be applied to SslSettings of the MongoClient Builder.
In case of exceptions caused by wrong certificates or passwords, or anything else, the error message will be presented in the logs, and those settings will be ignored and a connector eventually will fail on validation of a client (Handshake exception).

### CODE REVIEW

**Comment by bdesert** [comment]
@rozza, @jyemin ,
as discussed, I've implemented it this time by adding truststore and keystore that won't impact entire JVM, but only SSL Context of the MongoClient.
I've tested it on my environment and it looks good. let me know if you have any questions or suggestions.

Thanks.

**Review by rozza — APPROVED** [review]

File: src/main/java/com/mongodb/kafka/connect/util/SslConfigs.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+          keyManagers = keyManagerFactory.getKeyManagers();
+        }
+      }
+
+      if (trustManagers == null && keyManagers == null) {
+        // either key or trust managers can be null, and SSLConext can be updated with nulls
+        // in this case the installed security providers will
+        // be searched for the highest priority implementation of the appropriate factory.
+        // But, if SSL configuration parameters aren't provided (both are null),
+        // it's better to leave SSLConext without a change at all.
```
Comment (jyemin) [inline src/main/java/com/mongodb/kafka/connect/util/SslConfigs.java:173]: SSLConext => SSLContext

**Review by jyemin — APPROVED** [review]
One typo, but otherwise LGTM.

File: src/main/java/com/mongodb/kafka/connect/util/SslConfigs.java (hunk 1 above)
Comment (bdesert) [reply src/main/java/com/mongodb/kafka/connect/util/SslConfigs.java]: Fixed the typo.
thanks for catching this... 👍

**Comment by bdesert** [comment]
@rozza the merge didn't go well. the build will start failing now.

**Comment by rozza** [comment]
@bdesert apologies - missed that, I messed up my merge commit.  Updated in master.

**Pull request merged**